src/zh/zaimanhua/src/eu/kanade/tachiyomi/extension/zh/zaimanhua/ZaimanhuaDto.kt

```diff
@@ -0,0 +1,137 @@
+package eu.kanade.tachiyomi.extension.zh.zaimanhua
+
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.JsonNames
+
+@Serializable
+class MangaDto(
+    private val id: Int,
+    private val title: String,
+    private val cover: String,
+    private val description: String? = null,
+    private val types: List<TagDto>,
+    private val status: List<TagDto>,
+    private val authors: List<TagDto>,
+    @SerialName("chapters")
+    private val chapterGroups: List<ChapterGroupDto>,
+) {
+    fun toSManga() = SManga.create().apply {
+        url = id.toString()
+        title = this@MangaDto.title
+        author = authors.joinToString { it.name }
+        description = this@MangaDto.description
+        genre = types.joinToString { it.name }
+        status = parseStatus(this@MangaDto.status[0].name)
+        thumbnail_url = cover
+        initialized = true
+    }
+
+    fun parseChapterList(): List<SChapter> {
+        val mangaId = id.toString()
+        val size = chapterGroups.sumOf { it.size }
+        return chapterGroups.flatMapTo(ArrayList(size)) {
+            it.toSChapterList(mangaId)
+        }
+    }
+}
+
+@Serializable
+class ChapterGroupDto(
+    private val title: String,
+    private val data: List<ChapterDto>,
+) {
+    fun toSChapterList(mangaId: String): List<SChapter> {
+        val groupName = title
+        val isDefaultGroup = groupName == "连载"
+        return data.map {
+            it.toSChapterInternal().apply {
+                url = "$mangaId/$url"
+                if (!isDefaultGroup) name = "$groupName: $name"
```

Comment:
Multiple groups appear for a variety of reasons, sometimes to indicate different translators, sometimes to distinguish whether the source is a magazine or the web. 
I think it would be good to put it in SChapter.scanlator as well